src/state/controls/Control.spec.ts

```diff
@@ -58,4 +58,27 @@ describe('control', () => {
         });
         stub.restore();
     });
+
+    it('allows batch updates', () => {
+        const buttonDiff = {
+            cost: 200,
+            text: 'foobar',
+        };
+        const updatedButton = Object.assign(
+            {},
+            {
+                etag: buttonData.etag,
+                controlID: buttonData.controlID,
+            },
+            buttonDiff,
+        );
```

Comment:
It's guaranteed to be free of side effects, and is the standard we follow in all other TypeScript code